packages/playwright-test/src/runner/loadUtils.ts

```diff
@@ -129,19 +122,32 @@ export async function createRootSuite(testRun: TestRun, errors: TestError[], sho
   const config = testRun.config;
   // Create root suite, where each child will be a project suite with cloned file suites inside it.
   const rootSuite = new Suite('', 'root');
+  const projectSuites = new Map<FullProjectInternal, Suite>();
+  const filteredProjectSuites = new Map<FullProjectInternal, Suite>();
 
-  // First add top-level projects, so that we can filterOnly and shard just top-level.
+  // Filter all the projects using grep, testId, file names.
```

Comment:
Tests:
- setup+project+teardown have foo.spec, run "npx playwright test foo.spec" should run all teardown files
- `--grep` and `.only` inside setup file should both apply.
- `--grep` in setup file should not run chromium project.
- play in UI mode on setup test.